lib/networking/api_base_helper.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'api_exceptions.dart';
+import 'dart:async';
+
+class ApiBaseHelper {
+  final String _baseURL = "https://api.nookipedia.com/";
+
+  dynamic _returnResponse(http.Response response) {
+    switch (response.statusCode) {
+      case 200:
+        var responseJson = jsonDecode(response.body.toString());
+        print(responseJson);
+        return responseJson;
+      case 400:
+        throw BadRequestException(response.body.toString());
+      case 401:
+      case 403:
+        throw UnauthorisedException(response.body.toString());
+      case 500:
+      default:
+        throw FetchDataException(
+            'Error occured while Communication with Server with StatusCode : ${response.statusCode}');
+    }
+  }
+
+  Future<dynamic> get(String url) async {
+    print('Api get url $url');
+    var responseJson;
+    try {
+      final response = await http.get(Uri.parse(_baseURL + url));
```

Comment:
don't do `baseURL + url`
What if in future there are more apis to call? please make the method as abstract as possible, the method shouldn't care what API URL it's trying to get, make it as generic as possible